File: apps/web/repositories/api/home.repository.ts
// API Home Repository implementation
import type { IHomeRepository } from '../types';
import type { HomeOut, HomeCreate, HomeUpdate } from '@/types/api';
import { apiClient } from '@/services/api-client';

export class ApiHomeRepository implements IHomeRepository {
    public async list(): Promise<HomeOut[]> {
        return apiClient.get('/homes');
    }

    public async get(id: string): Promise<HomeOut> {
        return apiClient.get(`/homes/${id}`);
    }

    public async create(req: HomeCreate): Promise<HomeOut> {
        return apiClient.post('/homes', req);
    }

    public async update(id: string, req: HomeUpdate): Promise<HomeOut> {
        return apiClient.patch(`/homes/${id}`, req);
    }

    public async delete(id: string): Promise<void> {
        return apiClient.delete(`/homes/${id}`);
    }
}

File: apps/android/core/src/main/java/com/atech/core/data/HomeRepository.kt
package com.atech.core.data

import com.atech.core.common.DomusResult
import com.atech.core.model.Home
import com.atech.core.model.HomeCreate
import com.atech.core.model.HomeUpdate
import com.atech.core.network.DomusHttp
import com.atech.core.network.safeApiCall
import io.ktor.client.request.delete
import io.ktor.client.request.get
import io.ktor.client.request.patch
import io.ktor.client.request.post
import io.ktor.client.request.setBody

/** Homes. Mirrors `backend/homes/router.py`. */
class HomeRepository(private val http: DomusHttp) {
    private val base get() = "${http.config.apiBase}/homes"

    suspend fun list(): DomusResult<List<Home>> =
        safeApiCall { http.client.get(base) }

    suspend fun get(homeId: String): DomusResult<Home> =
        safeApiCall { http.client.get("$base/$homeId") }

    suspend fun create(data: HomeCreate): DomusResult<Home> =
        safeApiCall { http.client.post(base) { setBody(data) } }

    suspend fun update(homeId: String, data: HomeUpdate): DomusResult<Home> =
        safeApiCall { http.client.patch("$base/$homeId") { setBody(data) } }

    suspend fun delete(homeId: String): DomusResult<Unit> =
        safeApiCall { http.client.delete("$base/$homeId") }
}
